generators/static.py

```diff
@@ -0,0 +1,226 @@
+import sys
+from random import randint, uniform
+
+# Constants
+DEVICES = ["Desktop", "Phone", "Tablet"]
+COUNTRIES = ["Ukraine", "United States of America", "United Kingdom"]
+
+# Defaults
+ITEMS_TO_GENERATE_DEFAULT = 100_000
+MAX_ITEM_ID_DEFAULT = 1_000_000_000
+MIN_ITEM_ID_DEFAULT = 0
+MIN_ITEM_AMOUNT_DEFAULT = 0
+MAX_ITEM_AMOUNT_DEFAULT = 100_000
+MIN_ITEM_PRICE_DEFAULT = 0.5
+MAX_ITEM_PRICE_DEFAULT = 1000_000.0
+
+USERS_TO_GENERATE_DEFAULT = 100_000
+MAX_DEVICES_NUMBER_DEFAULT = 10
+MIN_DEVICES_NUMBER_DEFAULT = 1
+MIN_USER_ID_DEFAULT = 0
+MAX_USER_ID_DEFAULT = 1000_000
+
+# Implementation
+
+used_item_id = {-1}
+
+used_user_id = {-1}
+used_ip_address = {"0.0.0.0"}
+
+args = {}
+
+
+def generate_users_and_ips_file(filepath_users, filepath_ip, users_number=USERS_TO_GENERATE_DEFAULT, **kwargs):
+    """
+    Generates users and ip files
+    :param users_number: number of users to generate
+    :param filepath_users: path to a file with users
+    :param filepath_ip: path to a file with ips
+    :return: nothing
+    """
+    with open(filepath_users, "w+") as file_users, open(filepath_ip, "w+") as file_ips:
+        for i in range(users_number):
+            user = generate_user()
+            write_user(file_users, user)
+            write_devices(file_ips, user[1], user[2])
+
+
+def generate_items(filepath, items_number=ITEMS_TO_GENERATE_DEFAULT, **kwargs):
+    """
+    Generates items and writes them to file
+    :param items_number: number of items to generate
+    :param filepath: path to a file to write items in
+    :return: nothing
+    """
+    with open(filepath, "w+") as file:
+        for i in range(items_number):
+            write_item_to_file(generate_item(), file)
+
+
+def generate_user():
+    """
+    Generates user
+    :return: tuple that represents user
+    """
+    return generate_user_id(**args), generate_devices(**args), generate_country()
+
+
+def generate_user_id(min_user_id=MIN_USER_ID_DEFAULT, max_user_id=MAX_USER_ID_DEFAULT, **kwargs):
+    """
+    Generates user id
+    :param min_user_id: lowest possible user id
+    :param max_user_id: biggest possible user id
+    :return: unique user id
+    """
+    user_id = -1
+    while user_id in used_user_id:
+        user_id = randint(min_user_id, max_user_id)
+    used_user_id.add(user_id)
+    return "user" + str(user_id)
+
+
+def generate_devices(min_devices_number=MIN_DEVICES_NUMBER_DEFAULT, max_devices_number=MAX_DEVICES_NUMBER_DEFAULT,
+                     **kwargs):
+    """
+    Generates devices
+    :param min_devices_number: lowest possible number of devices
+    :param max_devices_number: biggest possible number of devices
+    :return: list of tuples that represent devices
+    """
+    devices = []
+
+    for i in range(randint(min_devices_number, max_devices_number)):
+        devices.append((DEVICES[randint(0, len(DEVICES) - 1)], generate_random_ip()))
+
+    return devices
+
+
+def generate_random_ip():
```

Comment:
refactor this function.
use string default variable e like IP = "0.0.0.0" instead ted of blocks = [0, 0, 0, 0]. Then u don't need to convert list each time to str.